services/shhext/mailservers/peerstore_test.go

```diff
@@ -0,0 +1,76 @@
+package mailservers
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/p2p"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/stretchr/testify/require"
+)
+
+func RandomeNode() (*enode.Node, error) {
+	pkey, err := crypto.GenerateKey()
+	if err != nil {
+		return nil, err
+	}
+	return enode.NewV4(&pkey.PublicKey, nil, 0, 0), nil
+}
+
+func TestUpdateResetsInternalStorage(t *testing.T) {
+	store := NewPeerStore()
+	r1, err := RandomeNode()
+	require.NoError(t, err)
+	r2, err := RandomeNode()
+	require.NoError(t, err)
+	store.Update([]*enode.Node{r1, r2})
+	require.True(t, store.Exist(r1.ID()))
+	require.True(t, store.Exist(r2.ID()))
+	store.Update([]*enode.Node{r2})
+	require.False(t, store.Exist(r1.ID()))
+	require.True(t, store.Exist(r2.ID()))
+}
+
+func TestGetNodeByID(t *testing.T) {
+	store := NewPeerStore()
+	r1, err := RandomeNode()
+	require.NoError(t, err)
+	store.Update([]*enode.Node{r1})
+	require.Equal(t, r1, store.Get(r1.ID()))
+	require.Nil(t, store.Get(enode.ID{1}))
+}
+
+type fakePeerProvider struct {
+	peers []*p2p.Peer
+}
+
+func (f fakePeerProvider) Peers() []*p2p.Peer {
+	return f.peers
+}
+
+func TestNoConnected(t *testing.T) {
+	provider := fakePeerProvider{}
+	store := NewPeerStore()
+	_, err := GetFirstConnected(provider, store)
+	require.EqualError(t, ErrNoConnected, err.Error())
+}
+
+func TestGetFirstConnected(t *testing.T) {
+	numPeers := 1
```

Comment:
Why not test with multiple values, e.g. `numPeers:=1` and `numPeers:=3`? Otherwise one could change the logic of `GetFirstConnected` to return the last connected and this test would still pass.

Reply:
it is probably should be renamed to `GetAnyConnected`. the idea is that caller doesn't care which mail server will be used for making a request. but sure, there is no reason not to test with multiple values